### DISCUSSION TITLE
do something about empty autogenerated sidebar groups

### DISCUSSION THREAD

**Discussion by boneskull** [withastro/starlight#1410] [Feature Requests]
### What version of `starlight` are you using?

0.16.0

### What version of `astro` are you using?

4.2.4

### What package manager are you using?

npm

### What operating system are you using?

mac

### What browser are you using?

vivaldi

### Describe the Bug

If the directory specified in the `directory` prop of an autogenerate-style sidebar item is either a) missing or b) empty, Starlight shows an empty, expandable group.

I'm not sure what should happen instead (refuse to build? just hide it? warn?), but the current behavior is undesirable.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-btsg8v?file=astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for the issue @boneskull! Not sure it’s a “bug” exactly — the config says to generate a group, Starlight generates the group as asked, it just happens not to have content yet.

Were there particular scenarios where you found yourself hitting this in particular? Would be good to understand the use case better.

**Reply by HiDeoo** [reply]
Just saw [this issue](https://github.com/HiDeoo/starlight-typedoc/issues/28) which is probably the case that was encountered here. I personally think this should be fixed at the plugin level too.

**Reply by delucis** [reply]
Ah OK, thanks for the extra context! In an automated tool, I guess it’s more common to accidentally generate empty groups. I agree it’s a bit odd to generate an empty group in the sidebar (it doesn’t really do anything), but on the whole I think I’d prefer keeping Starlight’s logic quite transparent so that what is authored in the config is reflected on the page (and then it’s up to users to make sure the config is correct).

**Reply by boneskull** [reply]
That makes enough sense to me. Thanks
